interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,218 @@
+    interface=core.ubuntu.media.Service
+    member="{Create,Detach,Reattach,Destroy,CreateFixed,Resume}Session"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to pause all other sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="PauseOtherSessions"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to query/modify service properties
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=org.freedesktop.DBus.Properties
+    member=PropertiesChanged
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var mediaHubConnectedSlotAppArmorClassic = []byte(`
+# Allow unconfined clients to interact with the player on classic
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    peer=(label=unconfined),
+`)
+
+var mediaHubConnectedPlugAppArmor = []byte(`
+# Description: Allow using media-hub service.
+
+#include <abstractions/dbus-strict>
+
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=system
+    path=/
+    interface=org.freedesktop.DBus.ObjectManager
+    peer=(label=unconfined),
+`)
+
+var mediaHubPermanentSlotSecComp = []byte(`
+recvmsg
+sendmsg
+sendto
+recvfrom
+`)
+
+var mediaHubConnectedPlugSecComp = []byte(`
+recvmsg
+sendmsg
+sendto
+recvfrom
+`)
+
+type MediaHubInterface struct{}
+
+func (iface *MediaHubInterface) Name() string {
+	return "media-hub"
+}
+
+func (iface *MediaHubInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		old := []byte("###SLOT_SECURITY_TAGS###")
+		new := []byte("")
+		if release.OnClassic {
+			// On classic MediaHub will run unconfined
+			new = []byte("unconfined")
+		} else {
+			new = slotAppLabelExpr(slot)
+		}
+		snippet := bytes.Replace(mediaHubConnectedPlugAppArmor, old, new, -1)
+		return snippet, nil
+	case interfaces.SecuritySecComp:
+		return mediaHubConnectedPlugSecComp, nil
+	}
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		return mediaHubPermanentSlotAppArmor, nil
+	case interfaces.SecuritySecComp:
+		return mediaHubPermanentSlotSecComp, nil
+	}
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
```

Comment:
ConnectedSlotSnippet should do the same thing for `###PLUG_SECURITY_TAGS###` and ConnectedSlotAppArmor that ConnectedPlugSnippet does for `###SLOT_SECURITY_TAGS###"` and ConnectedPlugAppArmor.

In other words, you need to define in ConnectedSlotSnippet the policy needed for the server to receive the plugs side  communication just like ConnectedPlugSnippet defines the policy needed for the client to send the slots side communication to the server.